packages/zowe-explorer/src/webviews/src/table-view/ActionsBar.tsx

```diff
@@ -1,66 +1,174 @@
-import { Ref } from "preact/hooks";
+import { Dispatch, Ref, useState } from "preact/hooks";
 import type { Table } from "@zowe/zowe-explorer-api";
-import { VSCodeButton } from "@vscode/webview-ui-toolkit/react";
+import { VSCodeButton, VSCodeTextField } from "@vscode/webview-ui-toolkit/react";
 import { GridApi } from "ag-grid-community";
+import { wrapFn } from "./types";
+import { FocusableItem, Menu, MenuGroup, MenuItem } from "@szhsin/react-menu";
+import "@szhsin/react-menu/dist/index.css";
 
 export const ActionsBar = ({
   actions,
   gridRef,
   itemCount,
+  selectionCount,
+  title,
+  columns,
+  visibleColumns,
+  setVisibleColumns,
   vscodeApi,
 }: {
   actions: Table.Action[];
   gridRef: Ref<any>;
   itemCount: number;
+  selectionCount: number;
+  title: string;
+  columns: string[];
+  visibleColumns: string[];
+  setVisibleColumns: Dispatch<string[]>;
   vscodeApi: any;
 }) => {
+  const [searchFilter, setSearchFilter] = useState<string>("");
+
+  const columnDropdownItems = (visibleColumns: string[]) =>
+    columns
+      .filter((col) => col !== "actions" && (searchFilter.length === 0 || col.toLowerCase().includes(searchFilter)))
+      .map((col) => (
+        <MenuItem
+          key={`toggle-vis-${col}`}
+          type="checkbox"
+          onClick={(e: any) => {
+            const gridApi = gridRef.current.api as GridApi;
+            const colVisibility = !visibleColumns.includes(col);
+            gridApi.setColumnsVisible(
+              [gridApi.getColumns()?.find((c) => c.getColDef().field === col || c.getColDef().headerName === col)!],
+              colVisibility
+            );
+            setVisibleColumns(colVisibility ? [...visibleColumns, col] : visibleColumns.filter((c) => c !== col));
+            e.keepOpen = true;
+          }}
+        >
+          {() => (
+            <div
+              style={{
+                display: "flex",
+                justifyContent: "space-evenly",
+                alignItems: "center",
+                height: "1.25rem",
+                marginLeft: "-1.25rem",
+                overflow: "hidden",
```

Comment:
This may apply to other style blocks (or one-liners)...
Could we consolidate some of the static stylings into defined css (or scss) classes? 😅